src/cairoUtilFuncGen/storage/dynArrayPushWithArg.ts

```diff
@@ -167,6 +167,9 @@ export class DynArrayPushWithArgGen extends StringIndexedFuncGen {
     });
     this.requireImport('starkware.cairo.common.uint256', 'Uint256');
     this.requireImport('starkware.cairo.common.uint256', 'uint256_add');
+    this.requireImport('starkware.cairo.common.cairo_builtins', 'BitwiseBuiltin');
+    this.requireImport('starkware.cairo.common.dict_access', 'DictAccess');
+    this.requireImport('starkware.cairo.common.cairo_builtins', 'HashBuiltin');
```

Comment:
The bug here is not missing imports, but the consequence of a bigger problem which is the function pruner not deleting generated functions (This applies to all generated warp functions). Which contract throwed this error? General pattern is:
```solidity
function generated_function() { ... }
function f() private { 
    generated_function()
}
```
`f` get's removed in function prunner but `generated_function` does not (It gets deleted, but since we handle generated functions printing different, it get's printed anyways). Then the implicit args don't get imported and so the error.

My suggested solution would be change part of the project structure and link generated function strings with their definitions (instead of handling separately). This is not completely straightforward to implement because there are some edge cases and there is a lot to change.

But I think for now a better patch would be non-deleting generated functions, since they'll definition will still exists, will produce cairo code with unused code (which we already do) but without the bugs

Reply:
Thanks for the explanation, this bug has reared its head a few times now. I think its the right time to tackle it properly maybe we make a new node for generated function definitions and attach the source code for the function in the node so that we can manipulate it properly. @rodrigo-pino how does this sound to you?

Reply:
Yes, I think that is the best approach, and I think  this new node could be used as well for cairo stubs, which we currently handle some times as solidity function and other times as cairo stubs (and I think it should be always handled as cairo stubs)